samples/iolib/src/main/cpp/player/SimpleMultiPlayer.cpp

```diff
@@ -72,13 +72,34 @@ void SimpleMultiPlayer::onErrorBeforeClose(AudioStream *, Result error) {
     __android_log_print(ANDROID_LOG_INFO, TAG, "==== onErrorBeforeClose() error:%d", error);
 }
 
+int SimpleMultiPlayer::getDeviceSampleRate(int32_t channelCount) {
```

Comment:
Are you opening a stream just to get the sample rate?
Can you open your real stream first before you load the samples?

Rapidly opening and closing streams can expose race conditions in Audio Policy and HAL that are dangerous. Also you will not get a low latency stream anyway if you do not use a callback.

Reply:
I had hoped it could just be this "atomic" query. I'll need to do a little rearchitecting to do as you suggest, but it seems like we need to.

Reply:
Did the required "rearchitecting".
Done.